### PR TITLE
rm verify_test_dsa.go

### PR DESCRIPTION
**Pull request by AkihiroSuda**
#### Name of feature:
Not a feature.

#### Pain or issue this feature alleviates:
- This test can be safely removed because DSA signature verification is no longer supported since [Go 1.16](https://go.dev/doc/go1.16) and Go <= 1.21 are [no longer supported](https://endoflife.date/go).
   The test have been actually skipped by `go test` because it doesn't have the `_test.go` suffix.
- The test was leaked into the `vendor` directory of projects consuming this package. The `var dsaPrivateKey = ...` line (L128) could cause confusion to some security scanning tools.

#### Why is this important to the project (if not answered above):
Answered above.

#### Is there documentation on how to use this feature? If so, where?
Not a feature.

#### In what environments or workflows is this feature supported?
Not a feature.

#### In what environments or workflows is this feature explicitly NOT supported (if any)?
Not a feature.

#### Supporting links/other PRs/issues:
N/A

💔Thank you!

### CODE REVIEW

File: verify_test_dsa.go [hunk 1]
```diff
@@ -1,182 +0,0 @@
...
-	}
-	if err := p7.Verify(); err != nil {
-		t.Fatalf("Verify failed with error: %v", err)
-	}
-	os.Remove(tmpSignerCertFile.Name()) // clean up
-	os.Remove(tmpSignerKeyFile.Name())  // clean up
-	os.Remove(tmpContentFile.Name())    // clean up
-}
-
-var dsaPrivateKey = []byte(`-----BEGIN PRIVATE KEY-----
```
Comment (AkihiroSuda) [inline verify_test_dsa.go:128]: This line was causing a confusion to key leakage detection tools

File: verify_test_dsa.go [hunk 2]
```diff
@@ -1,182 +0,0 @@
-// +build go1.11 go1.12 go1.13 go1.14 go1.15
```
Comment (AkihiroSuda) [inline verify_test_dsa.go:1]: This line was apparently intended to compile the test only on Go 1.11-1.15.
However, this was actually compiled for every Go (>= 1.11) releases.

The test was not executed by `go test` because the file does not have `_test.go` suffix.

**Comment by kiashok** [comment]
Thanks for opening this up @AkihiroSuda!

**Comment by kiashok** [comment]
@azazeal could you please take a look when you get a chance? Thanks!

**Review by azazeal — COMMENTED** [review]
I think that renaming the test file from `verify_test_dsa.go` to `verify_dsa_test.go` would probably negate the need to remove the test, as the `go.mod` specifies that the minimum required Go version is `1.14`.

Will also bring @hslatman in this.

**Comment by azazeal** [comment]
> I think that renaming the test file from `verify_test_dsa.go` to `verify_dsa_test.go` would probably negate the need to remove the test, as the `go.mod` specifies that the minimum required Go version is `1.14`.
> 
> Will also bring @hslatman in this.

I assume this would also clear up any confusion caused to security tools.

**Comment by AkihiroSuda** [comment]
> I think that renaming the test file from `verify_test_dsa.go` to `verify_dsa_test.go` would probably negate the need to remove the test, as the `go.mod` specifies that the minimum required Go version is `1.14`.

Why retain the test?
DSA signature verification is no longer supported since [Go 1.16](https://go.dev/doc/go1.16), and Go <= 1.21 are [no longer supported](https://endoflife.date/go).

**Comment by azazeal** [comment]
> > I think that renaming the test file from `verify_test_dsa.go` to `verify_dsa_test.go` would probably negate the need to remove the test, as the `go.mod` specifies that the minimum required Go version is `1.14`.
> 
> Why retain the test? DSA signature verification is no longer supported since [Go 1.16](https://go.dev/doc/go1.16), and Go <= 1.21 are [no longer supported](https://endoflife.date/go).

I understand that. What I'm trying to articulate is that the project's `go.mod` file specifies that the minimum required Go version is `1.14` ([ref](https://github.com/smallstep/pkcs7/blob/29d67d3d82eeaa810974229d4f8e80320fc2dde6/go.mod#L3)) and while I'm—100%—not advocating that compilation with said versions is a good idea anymore, the test does seem to make sense for those that still do (including our [CI](https://github.com/smallstep/pkcs7/blob/29d67d3d82eeaa810974229d4f8e80320fc2dde6/.github/workflows/ci.yml#L9)).

Renaming the file would allow us to keep the test that satisfies these edge cases, while at the same time clear up any confusion caused to the analysis tools.

**Comment by AkihiroSuda** [comment]
Thanks!

**Comment by hslatman** [comment]
> @hslatman @AkihiroSuda thanks for getting this in! Would you be able to cut a new release tag with this fix please?

It should already be in https://github.com/smallstep/pkcs7/releases/tag/v0.1.1, but do let me know if it doesn't fix the scanner issues.
